### PR TITLE
add support for `.rust-toolchain.toml`

### PR DESCRIPTION
**Pull request by TheAlgorythm**
This PR adds support for `.rust-toolchain.toml` files like described in #3182.
I'm not sure if this is the normal procedure or is a RFC needed?

### CODE REVIEW

**Comment by bjorn3** [comment]
cargo-crev will need to be updated too if this PR ends up getting accepted: https://github.com/crev-dev/cargo-crev/blob/5d9574d43634013ef73684f2af35f37db8ee8bfe/crev-lib/src/util/mod.rs#L118-L122

**Comment by ojeda** [comment]
`doc/src/overrides.md` here should be updated too.

File: doc/src/overrides.md [hunk 1]
```diff
@@ -77,9 +77,10 @@ case for nightly-only software that pins to a revision from the release
 archives.
 
 In these cases the toolchain can be named in the project's directory in a file
-called `rust-toolchain.toml` or `rust-toolchain`. If both files are present in
-a directory, the latter is used for backwards compatibility. The files use the
-[TOML] format and have the following layout:
+called `.rust-toolchain.toml`, `rust-toolchain.toml` or `rust-toolchain`.
+If multiple files are present in a directory, the latter present one is used
```
Comment (ojeda) [inline doc/src/overrides.md]: "the latter in the list" could clarify since there are more than 2 now.

**Review by ojeda — COMMENTED** [review]
Also: "The legacy format is not available in `rust-toolchain.toml` files." (in between the other two doc changes) could be updated to mention both files explicitly.

File: doc/src/overrides.md (hunk 1 above)
Comment (TheAlgorythm) [reply doc/src/overrides.md]: Maybe it would be better to use an enumeration for the precedence of the files.

Comment (TheAlgorythm) [reply doc/src/overrides.md]: Did the enumeration. Sorry for the bad phrasing in the first documentation attempt.

Comment (ojeda) [reply doc/src/overrides.md]: No worries at all! Thanks for working on this. :)

**Comment by rbtcollins** [comment]
So at this point both @kinnison and I dislike hiding the file. I'm not sure we can provide a sensible rational that can be argued through though. For now, since this would potentially affect all users and would not be a reversible decision, I think we should not merge a patch adding it.

**Comment by ojeda** [comment]
> So at this point both @kinnison and I dislike hiding the file. I'm not sure we can provide a sensible rational that can be argued through though.

Well, we provided a rational argument in the original issue, so perhaps you can find something there in particular that you do not like? Otherwise, it is hard to reply.

> For now, since this would potentially affect all users and would not be a reversible decision, I think we should not merge a patch adding it.

I agree a fair amount of people would probably start using it (which is one reason it would be nice to have! :), but it is backwards compatible, so I am not sure I understand how it would affect all users -- it should not affect any existing one, no?

**Pull request closed without merging**